Include/pystate.h

```diff
@@ -73,6 +73,20 @@ typedef int (*Py_tracefunc)(PyObject *, struct _frame *, int, PyObject *);
 #ifdef Py_LIMITED_API
 typedef struct _ts PyThreadState;
 #else
+
+typedef struct _excstate {
+    /* In a generator, we need to be able to swap between the exception
+       state inside the generator and the exception state of the calling
+       frame (which shouldn't be impacted when the generator "yields"
+       from an except handler).
+       These three fields exist exactly for that. */
+    PyObject *exc_type, *exc_value, *exc_traceback;
+
+    struct _excstate *exc_previous;
+
+} PyExcState;
```

Comment:
Would you consider a longer name here?  I always remember MvL saying "avoid abbreviations in identifiers, they are mystifying to non-English speakers".  "PyExceptionState" isn't that awful is it?

Reply:
`PyExcState` sounds ok to me.